src/include/sof/audio/module_adapter/module/generic.h

```diff
@@ -224,6 +232,8 @@ struct module_data {
 	struct module_interface *ops; /**< module specific operations */
 	struct module_memory memory; /**< memory allocated by module */
 	struct module_processing_data mpd; /**< shared data comp <-> module */
+	void *module_adapter; /**<loadable module interface handle */
+	uint32_t module_entry_point; /**<loadable module entry point address */
```

Comment:
You are modifying the core module_adapter code and also adding specific Intel code. Please split this commit into separate commits for each operation so that people can easily understand and review your change.

Reply:
OK, will split the commit.